sw/device/tests/sram_ctrl_execution_test_main.c

```diff
@@ -112,30 +83,42 @@ static bool otp_ifetch_enabled(void) {
 }
 
 /**
- * Performs SRAM execution test.
+ * Handles an exception.
  *
- * `ram_exec_function_t` is "mapped" onto the `execution_test_instructions`
- * buffer that holds hand-crafted machine instructions. Effectively, this
- * buffer becomes the function body.
+ * This exception handler only processes the faults that are relevant to this
+ * test. It falls into an infinite `wait_for_interrupt` routine (by calling
+ * `abort()`) for the rest.
  *
- * According to the calling convention the parameters will be passed in the
- * a0 and a1 registers, with a0 also treated as the return value register.
+ * The controlled fault originates in the retention SRAM, which means that
+ * normally the return address would be calculated relative to the trapped
+ * instruction. However, due to execution from retention SRAM being permanently
+ * disabled, this approach would not work.
+ *
+ * Instead the control flow needs to be returned to the caller. In other words,
+ * sram_execution_test -> retention_sram -> exception_handler
+ * -> sram_execution_test.
  */
-static void sram_execution_test(void) {
-  // Map the function pointer onto the instruction buffer.
-  sram_ctrl_testutils_exec_function_t func =
-      (sram_ctrl_testutils_exec_function_t)execution_test_instructions;
-
-  uintptr_t func_address = (uintptr_t)func;
-  CHECK(func_address >= kRamStartAddr && func_address <= kRamEndAddr,
-        "Test code resides outside of the Main SRAM: function address = %x",
-        func_address);
-
-  uint32_t expected_result = kExecParam1 + kExecParam2;
-  uint32_t result = func(kExecParam1, kExecParam2);
-  CHECK(result == expected_result,
-        "SRAM Execution expected %d + %d = %d, got %d", kExecParam1,
-        kExecParam2, expected_result, result);
+void handler_exception(void) {
+  uintptr_t ret_addr = (uintptr_t)OT_RETURN_ADDR();
+  LOG_INFO("Fault address: mepc = 0x%x, return address = 0x%x",
```

Comment:
Can you use `%08x` to zero-pad the pointers? Alternatively, use `%p`.

Reply:
Done.